starlette_context/middleware/raw_middleware.py

```diff
@@ -54,11 +61,31 @@ async def send_wrapper(message: Message) -> None:
 
         request = self.get_request_object(scope, receive, send)
 
-        _starlette_context_token: Token = _request_scope_context_storage.set(
-            await self.set_context(request)
-        )
+        try:
+            context = await self.set_context(request)
+        except ValueError as e:
+            if isinstance(e.args[0], Response):
+                error_response = e.args[0]
+            else:
+                error_response = self.error_response
+
+            message_head = {
+                "type": "http.response.start",
+                "status": error_response.status_code,
+                "headers": error_response.raw_headers,
+            }
+            await send(message_head)
```

Comment:
I know what it's supposed to do but I can't get use to this async weirdness. Can you link to another middleware that does a similar thing?